detach/detach_example_test.go

```diff
@@ -0,0 +1,175 @@
+package detach_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/reddit/baseplate.go/detach"
+)
+
+func init() {
```

Comment:
since `init` function is actually pretty special in go, I'm not sure if an `init` function inside a `_test.go` with `_test` package name will cause it to be executed when the package is imported. I'm 90% sure it should not, but it still makes me extra nervous.

I would prefer to split the 2 examples into 2 different files, and rename the `init` function into `InitInline`/`InitAsync` and call from the `Example*` function explicitly, with comment explaining that in real code they should just be named `init` without explicit call instead. This way the 2 examples will be rendered as full file examples so people can actually see how the init function is written (the current way pkg.go.dev will only render the `Example*` functions and hide everything else in this file).